internal/imageserver/imageserver.go

```diff
@@ -74,20 +76,19 @@ func (h *Handler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 		w.WriteHeader(http.StatusInternalServerError)
 		return
 	}
-
 	outPath, err := tempFileName(h.WorkDir)
 	if err != nil {
 		h.Log.WithError(err).Error("failed to create iso output file")
 		w.WriteHeader(http.StatusInternalServerError)
 		return
 	}
-	if err := create(outPath, isoWorkDir, lca_api.BlockDeviceLabel); err != nil {
-		h.Log.WithError(err).Error("failed to create iso")
+	defer os.Remove(outPath)
```

Comment:
`tempFileName` doesn't actually write anything to `outPath` so while moving this here is fine, it doesn't actually change anything.

Reply:
Whenever `create` fails an empty tempFile leaks and stay in the `/data/iso-workdir/`

Reply:
Ah right, because create makes the file itself. Good catch